pkg/protocol/commitment_verifier.go

```diff
@@ -78,9 +104,27 @@ func (c *CommitmentVerifier) verifyAttestations(attestations []*iotago.Attestati
 	var seatCount uint64
 
 	for _, att := range attestations {
-		// TODO: 1. Make sure the public key used to sign is valid for the given issuerID.
-		//  First, this can be based on the latest commonly known ledger state.
-		//  Later, this needs to include a proof of added/removed public keys.
+		// 1. Make sure the public key used to sign is valid for the given issuerID.
+		//    We ignore attestations that don't have a valid public key for the given issuerID.
+		//    1. The attestation might be fake.
+		//    2. The issuer might have added a new public key in the meantime, but we don't know about it yet
+		//       since we only have the ledger state at the forking point.
+		accountData, exists := c.validatorAccountsAtFork[att.IssuerID]
+
+		// We always need to have the accountData for a validator.
```

Comment:
... unless the attestation provided belongs to a different committee

Reply:
I agree, we should have a test of engine switching over epoch boundaries where a committee changes. However, this shouldn't be part of this PR. I've created #210 to keep track of this